Api v1 Beta Test Guide

ghubstan opened the pull request.

ghubstan committed:
Provide Api v1 Beta Test Guide

ghubstan committed:
Fix typo

ripcurlx commented:
@m52go Could you please proof-read the provided text?

m52go commented:
Working on it, will have it done by tomorrow.

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+Alternatively, you can specify any or all of these bisq-apitest options in a properties file located in
+`apitest/src/main/resources/apitest.properties`.
+
+In this example, a beta tester uses the `apitest.properties` below, instead of ./bisq-cli options.
```
```suggestion
In this example, a beta tester uses the `apitest.properties` below, instead of `./bisq-cli` options.
```

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+
+The regtest trade simulation script `apitest/scripts/trade-simulation.sh` is a useful introduction to the Bisq Api.
+The bash script’s output is intended to serve as a tutorial, showing how the CLI can be used to create payment
+accounts for Bob and Alice, create an offer, take the offer, and complete a trade protocol.
```
```suggestion
accounts for Bob and Alice, create an offer, take the offer, and complete a trade.
```

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+-c=<country>            The two letter country code, US, FR, AT, RU, etc.
+-f=<fixed-price>        The offer’s fixed price.
+    OR  (-f and -m options mutually exclusive, use one or the other)
+-m=<margin-from-price>  The offer’s margin (%) from market price.
```
```suggestion
-m=<margin-from-price>  The offer’s margin (%) below the current market price.
```

m52go replied:
Just realized this probably changes whether the offer is to buy or to sell. So disregard.

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+This simulation creates US / USD face-to-face payment accounts for Bob and Alice.  Alice (always the trade maker)
+creates a SELL / USD offer for the amount of 0.1 BTC, at a price 2% below the current market price.
+Bob (always the taker), will use his face-to-face account to take the offer, then the two sides will complete
+the trade protocol, checking their trade status along the way, and their BSQ / BTC balances when the trade is closed.
```
```suggestion
the trade, checking their trade status along the way, and their BSQ / BTC balances when the trade is closed.
```

m52go commented on apitest/docs/api-beta-test-guide.md:
````diff
@@ -0,0 +1,492 @@
+```
+$ ./bisq-cli --password=xyz --port=9998 getfundingaddresses
+```
+You can check block explorers for the status of a transaction, and check your Bisq BTC wallet address directly:
````
```suggestion
You can check a block explorer for the status of a transaction, or you can check your Bisq BTC wallet address directly:
```

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+
+Give the public address to the sender.  After the BSQ is sent, you can check block explorers for the status of
+the transaction.  There is no support (yet) to check the balance of an individual BSQ address in your wallet,
+but you can check your BSQ wallet’s balance to see when the new funds arrive:
```
```suggestion
but you can check your BSQ wallet’s balance to determine if the new funds have arrived:
```

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+
+### Sending BSQ and BTC to External Wallets
+
+Below are commands for sending BSQ and BTC to external wallets.
```
Sending BSQ to a non-Bisq wallet is not supported and is likely to cause problems and/or losses...maybe a warning like the following could be worth adding here?

"Please note that sending BSQ to non-Bisq wallets is not supported and highly discouraged."

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+### Withdrawal Transaction Fees
+
+If you have traded using the Bisq UI, you are probably aware of the default network bitcoin withdrawal transaction
+fee, and the custom withdrawal transaction fee user preference in the UI’s setting view.  The Api uses these same
```
```suggestion
If you have traded using the Bisq UI, you are probably aware of the default withdrawal transaction
fee and custom withdrawal transaction fee user preference in the UI’s setting view.  The Api uses these same
```

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+#### Examples
+
+The `trade-simulation.sh` script described above is an easy way to figure out how to use this command.
+In one of the examples, Alice created a BUY/ EUR offer to buy 0.125 BTC at a fixed price of  30,800 EUR,
```
```suggestion
In a previous example, Alice created a BUY/ EUR offer to buy 0.125 BTC at a fixed price of  30,800 EUR,
```

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+The `trade-simulation.sh` script described above is an easy way to figure out how to use this command.
+In one of the examples, Alice created a BUY/ EUR offer to buy 0.125 BTC at a fixed price of  30,800 EUR,
+and pay the Bisq maker fee in BSQ.  Alice had already created an EUR face-to-face payment account with id
+`f3c1ec8b-9761-458d-b13d-9039c6892413`, and used this `createoffer` command:
```
How does one know this payment account ID? Does `createpaymentacct` return it upon successful account creation?

ghubstan replied:
Yes, `createpaymentacct` returns the account ID, so does `getpaymentaccts`.

ghubstan replied:
I added an explanation in this section (commit cdf15676dbbcffe7aa383e9c9b7d872f9cde3eb0).

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+### Completing Trade Protocol
+
+The first step in the Bisq trade protocol is completed when a `takeoffer` command successfully creates a new trade from
+the taken offer.  After the Bisq servers prepare the trade, its status can be viewed with the `gettrade` command:
```
```suggestion
the taken offer.  After the Bisq nodes prepare the trade, its status can be viewed with the `gettrade` command:
```

I see how 'server' might make sense in this context but 'node' is more commonly-used throughout docs.

m52go commented on apitest/docs/api-beta-test-guide.md:
```diff
@@ -0,0 +1,492 @@
+Deposit Published           YES if the taker fee tx deposit has been broadcast to the network.
+Deposit Confirmed           YES if the taker fee tx deposit has been confirmed by the network.
+Fiat Sent                   YES if the buyer has sent a “payment started” message to seller.
+Fiat Received               YES if the sell has sent a “payment received” message to buyer.
```
```suggestion
Fiat Received               YES if the seller has sent a “payment received” message to buyer.
```

m52go commented on apitest/docs/api-beta-test-guide.md:
````diff
@@ -0,0 +1,492 @@
+withdrawfunds               Send trade proceeds to an external wallet.
+```
+The last two mutually exclusive commands (`keepfunds` or `withdrawfunds`) may seem unnecessary, but they are critical
+because they inform the server that a trade’s state can be set to `CLOSED`.  Please close out your trades with one or
````
```suggestion
because they inform the Bisq node that a trade’s state can be set to `CLOSED`.  Please close out your trades with one or
```

m52go reviewed (commented):
Great overall -- most suggestions I made are minor items that I think would help clarity.

Just one minor point of confusion about where the payment account ID comes from, which I noted in a [comment](https://github.com/bisq-network/bisq/pull/5159#discussion_r574762773).

ghubstan committed:
Edit guide as per suggested changes

This commit contains changes suggested in review:
https://github.com/bisq-network/bisq/pull/5159#pullrequestreview-588900434

ghubstan committed:
Add missing punctuation (.)

sqrrm reviewed (approved):
utACK
